elephant/test/test_signal_processing.py

```diff
@@ -15,12 +15,48 @@
 import scipy.stats
 from numpy.testing.utils import assert_array_almost_equal
 import quantities as pq
-
 import elephant.signal_processing
-
 from numpy.ma.testutils import assert_array_equal, assert_allclose
 
 
+class XCorrelationTestCase(unittest.TestCase):
+
+    def test_cross_correlation(self):
+        '''
+        Test cross-correlation function for two sinusoids
+        Note, that accuracy depends on N and min(f).
+        E.g., f=0.1 and N=2018 only has an accuracy on the order decimal=1
+        '''
+        # Sine with phase shift phi vs cosine for different frequencies
+        dt = 0.02
+        freq = np.linspace(0.5, 15, 8)
+        N = 2018
+        t = np.arange(N)*dt
+        phi = np.pi/6.
+        x = np.zeros((N,4))
+        for f in freq:
+            x[:,0] = 0.2 * np.sin(2.*np.pi*f*t+phi)
+            x[:,1] = 10.7 * np.sin(2.*np.pi*f*t+phi*2.)
+            x[:,2] = 5.3 * np.cos(2.*np.pi*f*t)
+            x[:,3] = 1875.35 * np.cos(2.*np.pi*f*t)
+            # Generate neo.AnalogSignals from x and y
+            signal = neo.AnalogSignal(x, units='mV', t_start=0.*pq.ms, 
+                                      sampling_rate=1/dt*pq.Hz, dtype=float)
+            rho, tau = elephant.signal_processing.cross_correlation_function(
+                    signal, [[0, 2], [1, 3]], dt=dt, nlags=N/4.6)
```

Comment:
No, it doesnt. I just wanted to use some float smaller than the number of input samples. Do you think it would be better to use a fixed float, e.g. 31.3? Or what would you suggest?

Reply:
It just looks like it has some deeper meaning if you write it like this. So you should either add a comment, telling that 4.6 is just a random choice, or use a fixed float, which you can also define together with all the other parameters above.

Reply:
I will define nlag = 30 in the beginning. I only use nteger in order not to complicate things